ci/test/04_install.sh

```diff
@@ -6,6 +6,53 @@
 
 export LC_ALL=C.UTF-8
 
+if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+
+  ${CI_RETRY_EXE} brew update
+
+  for i in $BREW_PACKAGES; do
+    for RETRIES in $(seq 1 3); do
+      echo "Upgrading ${i}"
+      if ! brew outdated | grep -q $i; then
+        echo "Already up to date or not installed."
+        break
+      fi
+
+      if brew upgrade $i; then
+        break
+      fi
+      echo "Attempt ${RETRIES} failed."
+      if [ "${RETRIES}" -eq 3 ]; then
+        echo "Too many retries. Aborting."
+        exit 1
+      else
+        echo "Retrying..."
+      fi
+    done
+
+    for RETRIES in $(seq 1 3); do
+      echo "Installing ${i}"
+      if brew list | grep -q $i || brew install $i; then
+        break
+      fi
+      echo "Attempt ${RETRIES} failed."
+      if [ "${RETRIES}" -eq 3 ]; then
+        echo "Too many retries. Aborting."
+        exit 1
+      else
+        echo "Retrying..."
+      fi
+    done
+
+  done
```

Comment:
homebrew :man_facepalming: 


This can't be written shorter, right?

Reply:
Probably, but I'm no bash-master. Homebrew doesn't have a "install or update if possible, don't complain otherwise" feature, because ... reason.

Reply:
Could you just run `brew update` then `brew install package-name --force` for each package you're trying to install?

Reply:
Reinstalling these packages at every Travis run seems very slow:
```
-f, --force
              Remove a previously cached version and re-fetch.
```

Reply:
I'd be fine with a solution that works, but isn't the fastest. We can optimize for speed in a follow up pull request.